src/charm.py

```diff
@@ -361,6 +374,53 @@ def set_main_unit(self, unit: str) -> None:
             logging.info("Setting main unit to be %s", unit)
             peer_relation[0].data[self.app].update({MAIN_UNIT_ID: unit})
 
+    def set_signing_key(self, signing_key: str) -> None:
+        """Create secret with signing key content.
+
+        Args:
+            signing_key: signing key as string.
+        """
+        peer_relation = self.model.relations[synapse.SYNAPSE_PEER_RELATION_NAME]
+        if not peer_relation:
+            logger.error(
+                "Failed to set signing key: no peer relation %s found",
+                synapse.SYNAPSE_PEER_RELATION_NAME,
+            )
+            return
+
+        if signing_key == self.get_signing_key():
+            logger.info("Received signing key but there is no change, skipping")
+            return
+        logger.debug("Adding signing key to secret: %s", signing_key)
+        secret = self.app.add_secret({"secret-signing-key": signing_key})
+        peer_relation[0].data[self.app].update({"secret-signing-id": typing.cast(str, secret.id)})
+
+    def get_signing_key(self) -> typing.Optional[str]:
+        """Get signing key from secret.
+
+        Returns:
+            Signing key as string or None if not found.
+        """
+        peer_relation = self.model.relations[synapse.SYNAPSE_PEER_RELATION_NAME]
+        if not peer_relation:
+            logger.error(
+                "Failed to get signing key: no peer relation %s found",
+                synapse.SYNAPSE_PEER_RELATION_NAME,
+            )
+            return None
+
+        secret_id = peer_relation[0].data[self.app].get("secret-signing-id")
+        if secret_id:
+            try:
+                secret = self.model.get_secret(id=secret_id)
+                logging.debug(secret.get_content().get("secret-signing-key"))
+                return secret.get_content().get("secret-signing-key")
+            except (ops.model.SecretNotFoundError, ValueError, TypeError) as exc:
+                logger.exception("Failed to get secret id %s: %s", secret_id, str(exc))
+                del peer_relation[0].data[self.app]["secret-signing-id"]
+                return None
```

Comment:
This `return` doesn't do anything as far as I can see

Reply:
Removed